# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1503675032 -7200
#      Fri Aug 25 17:30:32 2017 +0200
# Node ID 804d60784340ccf9f9a8a5677cddc5ec89bfb993
# Parent  d260f54d7dde9ad6c31b35c60c74397701ed8921
fix shortcuts for showing and giving focus to the variable editor

* main-window.cc (configure_shortcuts): set shortcut for action that switches
  to the variabel editor;
  (construct_window_menu): fix wrong name of menu entry for variable editor

* shortcut-manager.cc (do_init_data): initialize shortcuts related to the
  variable editor

diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- a/libgui/src/main-window.cc
+++ b/libgui/src/main-window.cc
@@ -2345,17 +2345,17 @@ main_window::construct_window_menu (QMen
 
   _editor_action = construct_window_menu_item
             (window_menu, tr ("Editor"), false, editor_window);
 
   _documentation_action = construct_window_menu_item
             (window_menu, tr ("Documentation"), false, doc_browser_window);
 
   _variable_editor_action = construct_window_menu_item
-            (window_menu, tr ("Documentation"), false, variable_editor_window);
+            (window_menu, tr ("Variable Editor"), false, variable_editor_window);
 
   window_menu->addSeparator ();
 
   _reset_windows_action = add_action (window_menu, QIcon (),
               tr ("Reset Default Window Layout"), SLOT (reset_windows ()));
 }
 
 void
@@ -2657,16 +2657,18 @@ main_window::configure_shortcuts ()
                                   "main_window:show_history");
   shortcut_manager::set_shortcut (_show_workspace_action,
                                   "main_window:show_workspace");
   shortcut_manager::set_shortcut (_show_file_browser_action,
                                   "main_window:show_file_browser");
   shortcut_manager::set_shortcut (_show_editor_action, "main_window:show_editor");
   shortcut_manager::set_shortcut (_show_documentation_action,
                                   "main_window:show_doc");
+  shortcut_manager::set_shortcut (_show_variable_editor_action,
+                                  "main_window:show_variable_editor");
   shortcut_manager::set_shortcut (_command_window_action, "main_window:command");
   shortcut_manager::set_shortcut (_history_action, "main_window:history");
   shortcut_manager::set_shortcut (_workspace_action,  "main_window:workspace");
   shortcut_manager::set_shortcut (_file_browser_action,
                                   "main_window:file_browser");
   shortcut_manager::set_shortcut (_editor_action, "main_window:editor");
   shortcut_manager::set_shortcut (_documentation_action, "main_window:doc");
   shortcut_manager::set_shortcut (_variable_editor_action, "main_window:variable_editor");
diff --git a/libgui/src/shortcut-manager.cc b/libgui/src/shortcut-manager.cc
--- a/libgui/src/shortcut-manager.cc
+++ b/libgui/src/shortcut-manager.cc
@@ -146,28 +146,32 @@ shortcut_manager::do_init_data ()
   init (tr ("Show File Browser"), "main_window:show_file_browser",
         prefix + ctrl_shift + Qt::Key_2);
   init (tr ("Show Workspace"), "main_window:show_workspace",
         prefix + ctrl_shift + Qt::Key_3);
   init (tr ("Show Editor"), "main_window:show_editor",
         prefix + ctrl_shift + Qt::Key_4);
   init (tr ("Show Documentation"), "main_window:show_doc",
         prefix + ctrl_shift + Qt::Key_5);
+  init (tr ("Show Variable Editor"), "main_window:show_variable_editor",
+        prefix + ctrl_shift + Qt::Key_6);
   init (tr ("Command Window"), "main_window:command",
         prefix + ctrl + Qt::Key_0);
   init (tr ("Command History"), "main_window:history",
         prefix + ctrl + Qt::Key_1);
   init (tr ("File Browser"), "main_window:file_browser",
         prefix + ctrl + Qt::Key_2);
   init (tr ("Workspace"), "main_window:workspace",
         prefix + ctrl + Qt::Key_3);
   init (tr ("Editor"), "main_window:editor",
         prefix + ctrl + Qt::Key_4);
   init (tr ("Documentation"), "main_window:doc",
         prefix + ctrl + Qt::Key_5);
+  init (tr ("Variable Editor"), "main_window:variable_editor",
+        prefix + ctrl + Qt::Key_6);
   init (tr ("Reset Default Window Layout"), "main_window:reset", QKeySequence ());
 
   // help
   init (tr ("Show Ondisk Documentation"), "main_help:ondisk_doc",
         QKeySequence ());
   init (tr ("Show Online Documentation"), "main_help:online_doc",
         QKeySequence ());
   init (tr ("Report Bug"), "main_help:report_bug", QKeySequence ());
